.github/workflows/helm-ci-test.yml

```diff
@@ -0,0 +1,101 @@
+name: Test helm installation
+on:
+  push:
+    branches:
+      - main
+
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test-go-get:
+    runs-on: ubuntu-latest
+    steps:
+      - name: '`go get` code'
+        run: |
+          # placeholder module so we can invoke go get
+          go mod init example.com/m
+
+          # note: cannot use github.sha here since that commit doesn't really exist
+          # however, since we require branches to always be up-to-date (GitHub setting),
+          # using the head of the PR branch should provide equivalent behaviour
+          sha='${{ github.event.pull_request.head.sha }}'
+          repo='${{ github.event.pull_request.head.repo.html_url}}'
+          repo=${repo#"https://"} # trim prefix
+
+          if [ "$repo" != "github.com/Azure/azure-service-operator" ]; then
+            echo "Skipping 'go get' check for fork…" 
+            exit 0
+          fi
```

Comment:
Under which circumstances would this skip the check? 

That is, when would the workflow be running and the branch NOT be ours?
(Suggest adding a comment to explain)

Reply:
This was to prevent check for forks. I've removed this one and as suggested by @matthchr, and have done it the other way to replace `controller:test-integration-kind` and `controller:test-integration-kind-ci` targets. Which resulted in not having any seperate workflow.